### PR TITLE
Feat/api testing harness

### LINKED ISSUE #275: Setup api test harness

### LINKED ISSUE #295: Run api tests via github actions on PR

### PR DESCRIPTION
**Pull request by mattyg**
Resolves #275 
Resolves #295 

Note that you will have to completely delete the mongodb volume and re-create it to setup the test database locally via the `users-init.sh` script. You can do so with: `yarn run mongodb:stop && docker volumes --prune && yarn run mongodb:start` 

This PR makes changes that will collide with a few others in the pipeline so will need to be merged carefully.

**Overview**
- refactor api server initialization logic to allow for arbitrary override of env configuration based on `NODE_ENV`
- modify `users_init.sh` database setup script to create an ephemeral testing database: `praise_db_testing_tmp`
- setup api test architecture
- add dev dependency `supertest`, `nyc`
- happy and sad path tests for auth endpoints
- refactor /api/auth endpoint to remove unused 'message' parameter
- Run api tests in github actions
- split up github actions to only run checks relevant to modified workspaces

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -17,3 +17,7 @@ jobs:
       - run: yarn workspace frontend build
       - run: yarn workspace discord-bot lint
       - run: yarn workspace discord-bot build
+      - run: echo "${{secrets.ROOT_ENV_TESTING }}" > .env
+      - run: echo "${{secrets.API_ENV_TESTING }}" > packages/api/.env
+      - run: yarn mongodb:start
+      - run: yarn workspace api test
```
Comment (Vyvy-vi) [inline .github/workflows/node.js.yml:23]: I'm wondering if we really need to run this test on every PR. We'd need to run this only when the API is altered, right?

**Review by Vyvy-vi — APPROVED** [review]
looks good to me

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (mattyg) [reply .github/workflows/node.js.yml:23]: yeah good point -- I should probably split up the github actions to only run relevant checks for modified workspaces.

Comment (mattyg) [reply .github/workflows/node.js.yml]: done

**Review by nebs-dev — COMMENTED** [review]
Looks good to me too!

**Pull request merged**